app/controllers/news_items_controller.rb

```diff
@@ -1,9 +1,17 @@
 class NewsItemsController < ApplicationController
   def index
-    @news_items = NewsItem.published.to_a
+    @news_items = news_item_scope.to_a
   end
 
   def show
-    @news_item = NewsItem.published.find(params[:id])
+    @news_item = news_item_scope.where(slug: params[:slug]).first ||
+                 news_item_scope.where(id: params[:id]).first ||
+                 raise(ActiveRecord::RecordNotFound)
```

Comment:
Sure. The reason why it's required is to still resolved old wordpress links (ex: if someone clicks on an post url from google)
I'll extract that logic into a tested and documented class method tonight or later today...